services/frontend-service/src/legacy-ui/ActionsCart/CheckoutDialog.tsx

```diff
@@ -32,21 +32,25 @@ import { Close } from '@material-ui/icons';
 import { ActionsCartContext } from '../App';
 import { BatchAction, GetOverviewResponse } from '../../api/api';
 import { useUnaryCallback } from '../Api';
+import { AuthTokenContext } from '../App/AuthContext';
 import { addMessageToAction, CartAction, hasLockAction, isNonNullable } from '../ActionDetails';
 
 export const callbacks = {
-    useBatch: (acts: BatchAction[], success?: () => void, fail?: () => void) =>
+    useBatch: (acts: BatchAction[], success?: () => void, fail?: () => void, authHeader: any = {}) =>
         useUnaryCallback(
             useCallback(
                 (api) =>
                     api
                         .batchService()
-                        .ProcessBatch({
-                            actions: acts,
-                        })
+                        .ProcessBatch(
```

Comment:
Why do we just add the header to `overview & batch` services, what about the other services? I'm not sure if this the best way to send the Auth header I mean why do we need to use the context and then add it to the request, we should have something like middleware that added this header to all requests by default 🤔

Reply:
We don't have a middleware right now, and as of now only two services are used. (everything else would be piped through batch service) 

checked with sven on this, unless there is more services incoming, he suggested keeping it like so

Reply:
> We don't have a middleware right now, and as of now only two services are used. (everything else would be piped through batch service)
> 
Hmm, this is not quite correct, you can use `useUnaryCallback` & `useObservable` to achieve that.

Reply:
Please see the changes here: https://github.com/freiheit-com/kuberpult/pull/285#discussion_r942546720

Reply:
use unarycallback doesn't have access to which api is called right now. it only has access to the api level, not the api.service.function level

Reply:
Yes, they don't need to now what is calling but they can pass more arguments to the callback they calling, please take a look at the changes, if you didn't get it we can have a call

Reply:
updated